lib/matplotlib/widgets.py

```diff
@@ -702,14 +702,18 @@ def __init__(
             f'marker{k}': v for k, v in {**defaults, **handle_style}.items()
         }
 
+        verts = np.zeros([5, 2])
+
         if orientation == "vertical":
             self.track = Rectangle(
                 (.25, 0), .5, 2,
                 transform=ax.transAxes,
                 facecolor=track_color
             )
             ax.add_patch(self.track)
-            self.poly = ax.axhspan(valinit[0], valinit[1], 0, 1, **kwargs)
+            verts = self._fill_verts(verts, valinit)
+            poly = Polygon(verts, **kwargs)
+            poly.set_transform(self.ax.get_yaxis_transform(which="grid"))
```

Comment:
with above suggestion you can do:

```suggestion
            poly_transform = self.ax.get_yaxis_transform(which="grid"))
```

and then create the polygon as below